test/library/characterization/test_resonator_spectroscopy.py

```diff
@@ -33,6 +33,22 @@
 )
 
 
+class MockIQBackendDefaults(MockIQBackend):
```

Comment:
Do you have any reason to prepare dedicated class for this? I'm fine with (re)adding `defaults()` method to the base `FakeOpenPulse2QV2` class as long as it is only used for the internal tests and we don't develop any further logic on top of this legacy method.

Reply:
Currently, `ResonatorSpectroscopy` is the only case in the package where we need to refer to `defaults()`, so here I did just a small override to `MockIQBackend` (which inherits from `FakeOpenPulse2QV2`)  to add `defaults()` back (if there was a simple way to do `MockIQBackend.defaults = XXX` for a single test then I would do that, but the ways I know to do that are not any cleaner than doing a subclass with one added method). This way we know that no other experiment is relying on `defaults()` without us realizing it. My hope is that we can add measure_freq_est/drive_freqs to a more proper place in qiskit-ibm-provider and then not do any tests against `defaults()`.

Reply:
> This way we know that no other experiment is relying on defaults() without us realizing it

Make sense. Let's keep the subclass as-is. Can you clarify this intention in the inline comments? I think that will prevent other developers from writing a piece of code with this feature. 

> if there was a simple way to do MockIQBackend.defaults = XXX for a single test then I would do that

Alternatively you can also define Mixin (e.g. `LegacyBackendMixin` that implements `configuration`, `properties`, `defaults`), but I don't think there is much difference. Maybe this could make the intention clearer.

Reply:
Comments added in 44cd94d810de9a5d6c962dedb322ed02871a7e75